testApi/src/main/scala/africa/absa/testing/scapi/rest/response/action/ExtractJsonResponseAction.scala

```diff
@@ -61,13 +63,21 @@ object ExtractJsonResponseAction extends ResponseActions {
 
     val action = fromString(responseAction.name.toLowerCase).getOrElse(None)
     action match {
-      case StringFromList =>
+      case StringFromList | StringFromJsonPath =>
         val cacheKey = responseAction.params("cacheKey")
-        val listIndex = responseAction.params("listIndex").toInt
-        val jsonKey = responseAction.params("jsonKey")
         val cacheLevel = responseAction.params("cacheLevel")
 
-        stringFromList(response, cacheKey, listIndex, jsonKey, cacheLevel)
+        action match {
+          case StringFromList =>
+            val jsonKey = responseAction.params("jsonKey")
+            val listIndex = responseAction.params("listIndex").toInt
+            stringFromList(response, cacheKey, listIndex, jsonKey, cacheLevel)
+
+          case StringFromJsonPath =>
+            val jsonPath = responseAction.params("jsonPath")
+            stringFromJsonPath(response, jsonPath, cacheKey, cacheLevel)
+        }
```

Comment:
```suggestion
        case StringFromList =>
          val (cacheKey, cacheLevel) = cachingFunctionYieldingTuple()
          val jsonKey = responseAction.params("jsonKey")
          val listIndex = responseAction.params("listIndex").toInt
          stringFromList(response, cacheKey, listIndex, jsonKey, cacheLevel)

        case StringFromJsonPath =>
          val (cacheKey, cacheLevel) = cachingFunctionYieldingTuple()
          val jsonPath = responseAction.params("jsonPath")
          stringFromJsonPath(response, jsonPath, cacheKey, cacheLevel)
        }
```

Instead of matching twice just like this, rather match once and repeat one method call common for specific cases.

Reply:
Thank you for an improvement idea.
Fixed in commit 66e35c3c68c28063ee6086a1f587c0063a5a6929.